main.go

```diff
@@ -100,3 +115,66 @@ func renderPage(page string) string {
 	}
 	return result
 }
+
+type pageCache struct {
+	diskv *diskv.Diskv
+}
+
+// CachedPage is a page stored in the diskv cache.
+type CachedPage struct {
+	// The page content.
+	Page string
+	// Unix time (seconds).
+	CreatedAt int64
+}
+
+func (c *CachedPage) isValid(now int64) bool {
+	return now-c.CreatedAt < 60*60*24 // 1 day.
+}
+
+func newCache() *pageCache {
+	currentUser, err := user.Current()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "Could not read home directory: %v\n", err)
+		os.Exit(-1)
+	}
+
+	return &pageCache{
+		diskv: diskv.New(diskv.Options{
+			BasePath:     filepath.Join(currentUser.HomeDir, ".tldr", "cache"),
+			Transform:    func(key string) []string { return []string{} },
+			CacheSizeMax: 1024 * 1024,
+		})}
+}
+
+func (c *pageCache) write(command string, page string) {
+	var cacheBuffer bytes.Buffer
+	encoder := gob.NewEncoder(&cacheBuffer)
+	encoder.Encode(CachedPage{
+		CreatedAt: time.Now().Unix(),
+		Page:      page,
+	})
+	c.diskv.Write(command, cacheBuffer.Bytes())
+}
+
+func (c *pageCache) read(command string) (string, error) {
```

Comment:
I don't have a better alternative in mind, but the cache miss and real error conditions are being coalesced unnecessarily. If this wasn't a cache (expected to have misses) we would treat misses differently than (perhaps non recoverable?) errors.
Perhaps the cache could be stateful and handle its own errors, while its api giving clear distinctions on misses / errors? wdyt?